tree/ntuple/v7/doc/architecture.md

```diff
@@ -0,0 +1,28 @@
+RNTuple Code Architecture
+=========================
+
+> This document is meant for ROOT developers. It provides background information on the RNTuple code design and behavior.
+
+> This document is currently a stub.
+
+
+Semantics of reading a non-trivial objects
+==========================================
+
+Reading an object with RNTuple should be seen as _overwriting_ its persistent data members.
+Given a properly constructed and valid object, the object must ensure that it stays valid when overwriting its persistent data members.
+However, the object should not rely on its transient state to remain unchanged during reading:
+it may be destructed and constructed again when it is read as part of a collection (see below).
```

Comment:
As a side note, users have requested (or have enforced by replacing the objects at each entry) to be able to force the calling of the constructor/destructor.  This can be useful in lieu of properly reset (via a rule) the transient.   This could also relates to the case where the user class has no explicit constructor but some transients. (it looks like the code currently enforce the non-trivial case in that situation).